conductr_cli/test/test_conduct_agents.py

```diff
@@ -0,0 +1,145 @@
+from conductr_cli.test.cli_test_case import CliTestCase, strip_margin
+from conductr_cli import conduct_agents, logging_setup
+from conductr_cli.http import DEFAULT_HTTP_TIMEOUT
+from unittest.mock import patch, MagicMock
+
+
+class TestConductAgentsCommand(CliTestCase):
+    conductr_auth = ('username', 'password')
+    server_verification_file = MagicMock(name='server_verification_file')
+
+    default_url = 'http://127.0.0.1:9005/v2/agents'
+
+    default_args = {
+        'dcos_mode': False,
+        'scheme': 'http',
+        'host': '127.0.0.1',
+        'port': 9005,
+        'base_path': '/',
+        'api_version': '2',
+        'verbose': False,
+        'quiet': False,
+        'long_ids': False,
+        'conductr_auth': conductr_auth,
+        'server_verification_file': server_verification_file,
+        'role': None
+    }
+
+    fake_output = """
+        [
+          {
+            "address": "akka.tcp://conductr-agent@192.168.10.1:2552/user/reaper/cluster-client#-596247188",
+            "observedBy": [
+              {
+                "node": {
+                  "address": "akka.tcp://conductr@192.168.10.2:9004",
+                  "uid": "1129598726"
+                }
+              }
+            ],
+            "roles": [
+              "web"
+            ]
+          },
+          {
+            "address": "akka.tcp://conductr-agent@192.168.10.2:2552/user/reaper/cluster-client#-775189131",
+            "observedBy": [
+              {
+                "node": {
+                  "address": "akka.tcp://conductr@192.168.10.2:9004",
+                  "uid": "1129598726"
+                }
+              }
+            ],
+            "roles": [
+              "web"
```

Comment:
Can we try multiple roles here so we can assert the role matching logic and the output as well?